clang/include/clang/AST/Decl.h

```diff
@@ -5178,6 +5179,29 @@ class HLSLBufferDecl final : public NamedDecl, public DeclContext {
   friend class ASTDeclWriter;
 };
 
+class HLSLRootSignatureDecl final : public NamedDecl {
+  ArrayRef<llvm::hlsl::rootsig::RootElement> RootElements;
```

Comment:
This should probably be a `TrailingObjects` instead of an `ArrayRef` so that it is allocated with the Decl (and in adjacent cache-friendly memory).